packages/contracts/src/DefiLlamaOracle.sol

```diff
@@ -33,17 +45,31 @@ contract DefiLlamaOracle is TrustedOracle {
     ///       the DefiLlama query. The specification is a structured JSON that describes which
     ///       data to bring on-chain from DefiLlama and how, so that the answerer bot can act
     ///       as expected.
+    ///     - `uint256 measurementTimestamp`: a timestamp that indicates when the final measurement
+    ///       should be taken and posted on-chain.
     function _initialize(InitializeOracleParams memory _params) internal override {
-        string memory _specification = abi.decode(_params.data, (string));
+        (string memory _specification, uint256 _measurementTimestamp) = abi.decode(_params.data, (string, uint256));
 
         if (bytes(_specification).length == 0) revert InvalidSpecification();
+        if (_measurementTimestamp < block.timestamp + minimumElapsedTime) revert MeasurementTimestampTooClose();
+        if (_measurementTimestamp > IKPIToken(_params.kpiToken).expiration()) {
+            revert MeasurementTimestampAfterKPITokenExpiration();
+        }
+
         specification = _specification;
+        measurementTimestamp = _measurementTimestamp;
+    }
+
+    /// @dev Internal finalization logic. This simply checks if the measurement timestamp
+    /// has passed in order to validate if a finalization request is valid or not
+    function _finalize(uint256) internal view override {
```

Comment:
Should this function have the `override`? I'm having an error when building the contracts:

```
Function has override specified but does not override anything.
```